From 353060fb69f9687cf95bb7514fa695d926e0801a Mon Sep 17 00:00:00 2001
From: Sravanthi Palakonda <srapal@codeaurora.org>
Date: Mon, 31 Aug 2015 19:37:10 +0530
Subject: [PATCH] WifiStateMachine: Disable IpReachabilityMonitor feature through a config param

IPreachabilityMonitor feature results in Wi-Fi disconnection. Thus,
disable this feature through a configuration parameter.

Feature can be enabled via FB config :
https://github.com/CyanogenMod/android_frameworks_base/blob/26079d3a0f/core/res/res/values/config.xml#L2484

Change-Id: I9a76a1a8f77762b524b323141c62aa8708ae70ea
CRs-Fixed: 899312
Signed-off-by: Zdrowy Gosciu <ZdrowyGosciu+GITHUB@gmail.com>
---

diff --git a/service/java/com/android/server/wifi/WifiStateMachine.java b/service/java/com/android/server/wifi/WifiStateMachine.java
index 1c06e71..a3f6ec5 100644
--- a/service/java/com/android/server/wifi/WifiStateMachine.java
+++ b/service/java/com/android/server/wifi/WifiStateMachine.java
@@ -8536,7 +8536,8 @@
             // cause the roam to faile and the device to disconnect
             clearCurrentConfigBSSID("L2ConnectedState");
 
-            try {
+            if (mIsWiFiIpReachabilityEnabled) {
+              try {
                 mIpReachabilityMonitor = new IpReachabilityMonitor(
                         mContext,
                         mInterfaceName,
@@ -8546,8 +8547,9 @@
                                 sendMessage(CMD_IP_REACHABILITY_LOST, logMsg);
                             }
                         });
-            } catch (IllegalArgumentException e) {
+              } catch (IllegalArgumentException e) {
                 Log.wtf("Failed to create IpReachabilityMonitor", e);
+              }
             }
         }
 
